### PR TITLE
Visual dicom browser enh

### PR DESCRIPTION
**Pull request by Punzo**

### CODE REVIEW

**Comment by Punzo** [comment]
@lassoan new integrative PR from https://github.com/commontk/CTK/pull/1217

File: Libs/DICOM/Core/ctkDICOMDatabase_p.h [hunk 1]
```diff
@@ -174,23 +174,25 @@ class CTK_DICOM_CORE_EXPORT ctkDICOMDatabasePrivate
...
-  bool insertPatient(const ctkDICOMItem& dataset,
+  int insertPatient(const ctkDICOMItem& dataset,
     const QString& patientID,
     const QString& patientsName,
     int& databasePatientID);
-  bool insertConnectionName(const int& dbPatientID,
+  int insertConnectionName(const int& dbPatientID,
     const QString& connectionName);
-  bool updateConnections(const QString& dbPatientID,
+  int updateConnections(const QString& dbPatientID,
```
Comment (lassoan) [inline Libs/DICOM/Core/ctkDICOMDatabase_p.h:190]: Move out this "update" method from the list of "insert" methods and update its documentation.

File: Libs/DICOM/Core/ctkDICOMDatabase.cpp [hunk 2]
```diff
@@ -425,35 +424,34 @@ bool ctkDICOMDatabasePrivate::insertConnectionName(const int& dbPatientID,
...
-
   QSqlQuery updateConnectionsStatement(this->Database);
   updateConnectionsStatement.prepare("UPDATE Patients SET Connections = :connectionsData WHERE UID = :uid");
   updateConnectionsStatement.bindValue(":connectionsData", connectionsData);
   updateConnectionsStatement.bindValue(":uid", dbPatientID);
 
   if (!loggedExec(updateConnectionsStatement))
   {
-    return false;
+    return -1;
```
Comment (lassoan) [inline Libs/DICOM/Core/ctkDICOMDatabase.cpp:439]: This method returns -1 or true now. It is probably better if it returns a bool (success) instead.

File: Libs/DICOM/Core/ctkDICOMDatabase.cpp [hunk 3]
```diff
@@ -803,11 +801,16 @@ bool ctkDICOMDatabasePrivate::insertPatientStudySeries(const ctkDICOMItem& datas
   else
   {
     logger.debug("Insert new patient if not already in database: " + patientID + " " + patientsName);
-    if (this->insertPatient(dataset, patientID, patientsName, dbPatientID))
+    int patientMetadataInsertSuccess = this->insertPatient(dataset, patientID, patientsName, dbPatientID);
```
Comment (lassoan) [inline Libs/DICOM/Core/ctkDICOMDatabase.cpp:804]: "success" is a `bool` type, not `int`. The `-1` and `1` act like magic numbers here. Two cleaner options could be:
- Option A: `int insertedPatientsCount = this->insertPatient...` and then `insertedPatientsCount>0`.
- Option B: use an enum to distinguish between inserted/not-inserted/failed.
- Option C: add additional arguments that can return more detail.

File: Libs/DICOM/Core/ctkDICOMDatabase.cpp [hunk 4]
```diff
@@ -305,10 +305,10 @@ QStringList ctkDICOMDatabasePrivate::filenames(QString table)
 }
 
 //------------------------------------------------------------------------------
-bool ctkDICOMDatabasePrivate::insertPatient(const ctkDICOMItem& dataset,
-                                            const QString& patientID,
-                                            const QString& patientsName,
-                                            int& dbPatientID)
+int ctkDICOMDatabasePrivate::insertPatient(const ctkDICOMItem& dataset,
```
Comment (lassoan) [inline Libs/DICOM/Core/ctkDICOMDatabase.cpp:308]: returns !connectionNameFound, which is bool

File: Libs/DICOM/Core/ctkDICOMDatabase.cpp [hunk 5]
```diff
@@ -425,35 +424,34 @@ bool ctkDICOMDatabasePrivate::insertConnectionName(const int& dbPatientID,
...
-    return false;
+    return -1;
   }
 
   return true;
 }
 
 //------------------------------------------------------------------------------
-bool ctkDICOMDatabasePrivate::insertStudy(const ctkDICOMItem& dataset, const int& dbPatientID)
+int ctkDICOMDatabasePrivate::insertStudy(const ctkDICOMItem& dataset, const int& dbPatientID)
```
Comment (lassoan) [inline Libs/DICOM/Core/ctkDICOMDatabase.cpp:446]: can return false

**Comment by Punzo** [comment]
@lassoan thanks for the review. I have cleaned the dirty modification to the insert methods by using an enum:

`ctkDICOMDatabase::InsertResult`

**Review by lassoan — APPROVED** [review]
It looks good, thank you.

**Comment by Punzo** [comment]
> It looks good, thank you.

thanks for reviewing and merging!

**Pull request merged**